File: 백준/Bronze/1924. 2007년/2007년.cc
#include <iostream>
using namespace std;
int main(void){
    int x,y, days = 0;
    cin >> x >> y;
    days += y;
    while(--x){
        switch(x){
            case 1:case 3:case 5:case 7: case 8: case 10: case 12:
                days += 31;
                break;
            case 2:
                days += 28;
                break;
            default :       
                days += 30;     
                break;
        }
    }
    switch(days%7){
        case 1:
            cout << "MON"<< '\n';
            break;
        case 2:
            cout << "TUE" << '\n';
            break;
        case 3:
            cout << "WED" << '\n';
            break;
        case 4:
            cout << "THU" << '\n';
            break;
        case 5:
            cout << "FRI" << '\n';
            break;
        case 6:
            cout << "SAT" << '\n';
            break;
        case 0:
            cout << "SUN" << '\n';
            break;
    }
    return 0;
}

File: 백준/Bronze/1924. 2007년/2007년.java
import java.io.*;
import java.util.*;

public class Main {
	public static String[] yoil = {"MON", "TUE", "WED", "THU", "FRI", "SAT", "SUN"};
	public static int[] days = {0,31,28,31,30,31,30,31,31,30,31,30,31};
	public static void main(String[] args) throws IOException{
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine(), " ");
		int mon = Integer.parseInt(st.nextToken());
		int day = Integer.parseInt(st.nextToken());
		
		int totaldays = 0;
		for(int i = 1; i <= mon-1; i++) {
			totaldays += days[i];
		}
		totaldays += day-1; //1월 1일부터시작하므로
		
		System.out.println(yoil[totaldays%7]);
		
	}
}
